cmdutil/debug/debug.go

```diff
@@ -31,25 +33,72 @@ import (
 // Connect to the debug server with gops:
 //
 //	gops stack localhost:PORT
-func New(l logrus.FieldLogger, port int) *Server {
-	return &Server{
+func New(l logrus.FieldLogger, config Config) *Server {
+	server := &Server{
 		logger: l,
-		addr:   fmt.Sprintf("127.0.0.1:%d", port),
+		addr:   fmt.Sprintf("127.0.0.1:%d", config.Port),
 		done:   make(chan struct{}),
 	}
+	if config.Enabled {
+		server.pprof = NewPProfServer(l, &config.PProf)
+	}
+	return server
 }
 
 // Server wraps a gops server for easy use with oklog/group.
 type Server struct {
 	logger logrus.FieldLogger
 	addr   string
 	done   chan struct{}
+	pprof  *PProfServer
 }
 
 // Run starts the debug server.
 //
-// It implements oklog group's runFn.
+// It implements oklog group's runFn and pprof.
 func (s *Server) Run() error {
+
+	var wg sync.WaitGroup
+	gopsErrChan := make(chan error, 1)
+	pprofErrChan := make(chan error, 1)
```

Comment:
Do we need two done channels?

Reply:
We are starting two servers one for gops agent and one for pprof. 
So I have added two error channels for the two go routines

Reply:
You don't need these error channels at all because you are using the `sync.WaitGroup`.